growth/too/models.py

```diff
@@ -727,6 +728,169 @@ def flat(self):
+            return -1
+        cnt = 0
+        for ii, observation in enumerate(observation_list):
+            if cnt == 0:
+                tt = observation.obstime
+            else:
+                tt = min(observation.obstime, tt)
+            cnt = cnt + 1
+        return tt
+
+    def observation_totaltime(self, telescope, filt, start_time, end_time):
+        observation_list = self.get_observations(filt, start_time, end_time)
+        tot = 0
+        observation_ids = []
+        for ii, observation in enumerate(observation_list):
+            if observation.observation_id not in observation_ids:
+                tot = tot + observation.exposure_time/60.0
+                observation_ids.append(observation.observation_id)
+        return tot
+
+    def observation_limmag(self, telescope, filt, start_time, end_time):
+        observation_list = self.get_observations(filt, start_time, end_time)
+        limmags = []
+        for ii, observation in enumerate(observation_list):
+            if observation.limmag is not None:
+                limmags.append(observation.limmag)
+        return np.median(limmags)
+
+
+    def observation_probability(self, telescope, filt, start_time, end_time):
+        # localization = Localization.query.filter_by(
+        #     dateobs=self.dateobs,
+        #     localization_name=Localization.localization_name).one()
+
+        ipix = np.asarray(list(self.observation_ipix(filt, start_time, end_time)))
+        if len(ipix) > 0:
+            return self.flat_2d[ipix].sum()
+        else:
+            return 0.0
+
+    def get_observations(self, telescope, filt, start_time, end_time):
+
+        # localization = Localization.query.filter_by(
+        #     dateobs=self.dateobs,
+        #     localization_name=self.localization_name).one()
+        prob = self.flat_2d
+
+        # fields = Field.query.filter_by(telescope=telescope).all()
+        # field_prob_1, field_ids_1 = [], []
+        # field_prob_2, field_ids_2 = [], []
+        # for field in fields:
+        #     if field.field_id > 800 and telescope == 'ZTF':
+        #         field_ids_2.append(field.field_id)
+        #         field_prob_2.append(np.sum(prob[field.ipix]))
+        #     else:
+        #         field_ids_1.append(field.field_id)
+        #         field_prob_1.append(np.sum(prob[field.ipix]))
+
+        fields = Field.query.filter_by(telescope=telescope)
+
+        # calculate the overlap between the field ipix and the ipix corresponding to the prob in the 90 %
+        
+        # a list containing the credible level corresponding to each ipix
+        levels = self.credible_levels_2d()
+        idx = levels <= 0.90
+        # all of the ipix corresponding to the 90% region
+        localization_ipix = range(len(prob))[idx]
+        # select the fields that overlap with the ipix array; select fields where ipix && 90% ipix; && indicates overlap between tables
+
+        for field in fields:
```

Comment:
The `Field.overlap` method is undefined. Here's a pure Python implementation:

```python
field_ids = {field.field_id for field in fields if not set(field.ipix).isdisjoint(localization_ipix)}
```

Here's an SQLAlchemy implementation (possibly faster):

```python
query = models.Field.query.filter_by(
    telescope=telescope
).filter(
    models.Field.ipix.overlaps(localization_ipix)
)
field_ids = {field.field_id for field in query}
```